### PR TITLE
Move tests to github actions

### PR DESCRIPTION
**Pull request by alexander-schranz**
See https://github.com/alexander-schranz/jackalope-doctrine-dbal/pull/1/checks

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                include:
+                    - php-version: '7.3'
+                      dependencies: lowest
+                      db: sqlite
+                    - php-version: '7.3'
+                      dependencies: lowest
+                      db: mysql
+                    - php-version: '7.3'
+                      dependencies: lowest
+                      db: sqlite
```
Comment (dbu) [inline .github/workflows/test-application.yaml:43]:
```suggestion
                      db: psql
```

File: .github/workflows/test-application.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  tools: 'composer:v2'
+
+            - name: Install dependencies with Composer
+              uses: ramsey/composer-install@v1
+              with:
+                  dependency-versions: ${{ matrix.dependencies }}
+                  composer-options: --prefer-dist --no-suggest
```
Comment (dbu) [inline .github/workflows/test-application.yaml:77]:
```suggestion
                  composer-options: --prefer-dist
```

**Review by dbu — COMMENTED** [review]
can you please check the build in your fork: https://github.com/alexander-schranz/jackalope-doctrine-dbal/actions

it seems the sqlite driver is not installed with php 7.3 and mysql / psql seem to not be set up properly. maybe it is just the commands to create the database and initialize things, but maybe it is something more.

File: .github/workflows/test-application.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        steps:
+            - name: Checkout project
+              uses: actions/checkout@v2
+
+            - name: Install and configure PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  extensions: "${{ matrix.extension }}"
```
Comment (dbu) [inline .github/workflows/test-application.yaml:72]: this seems to not work, 7.2 build with sqlite still fails.

**Review by dbu — COMMENTED** [review]
the mysql and postgres builds don't work. it looks like something is not properly set up

**Comment by alexander-schranz** [comment]
@dbu need to have a deeper look at this, why this fails now with ubuntu 20 🙈

**Comment by alexander-schranz** [comment]
Damn did pitfall again into localhost vs 127.0.0.1 in newer envs 🙈 . Now the tests should be [green](https://github.com/alexander-schranz/jackalope-doctrine-dbal/pull/1/checks).

File: .github/workflows/test-application.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    push:
+        branches:
+            - 'master'
+            - '[0-9]+.x'
+            - '[0-9]+.[0-9]+'
+            - '[0-9]+.[0-9]+.x'
+
+jobs:
+    test:
+        name: 'PHP ${{ matrix.php-version }}, Database ${{ matrix.db }}, Dependencies ${{ matrix.dependencies }}'
```
Comment (dbu) [inline .github/workflows/test-application.yaml:14]:
```suggestion
        name: 'PHP ${{ matrix.php-version }}, Database ${{ matrix.db }} ${{ matrix.dependencies }}'
```

I'd like to shorten this as it is not readable even on a large screen

**Review by dbu — APPROVED** [review]
awesome, cheers!

**Pull request merged**